ddl/ddl_api.go

```diff
@@ -816,7 +846,7 @@ func checkDefaultValue(ctx sessionctx.Context, c *table.Column, hasDefaultValue
 		return nil
 	}
 
-	if c.GetDefaultValue() != nil {
+	if c.GetDefaultValue() != nil && !c.DefaultIsExpr {
```

Comment:
If you skip this code, you should check whether the data type is `int` or `bigint` by yourself.

Reply:
good catch~